tensorflow/compiler/xla/mlir_hlo/lib/Dialect/mhlo/IR/hlo_ops.cc

```diff
@@ -7630,16 +7630,78 @@ void SortOp::getCanonicalizationPatterns(RewritePatternSet& results,
 // TransposeOp
 //===----------------------------------------------------------------------===//
 
+template <typename T>
+static DenseElementsAttr foldTransposeHelper(mhlo::TransposeOp op,
```

Comment:
I think it's better to call this `foldTransposeImpl` as it implements all the logic.